src/library/notehead.lua

```diff
@@ -1,64 +1,261 @@
 local library = require("library.general_library")
 
 local config = {
-    diamond_open = 79,
-    diamond_closed = 79, -- per Elaine Gould, use open diamond even on closed regular notes, but allow it to be overridden
-    diamond_resize = 110,
-    diamond_whole_offset = 5,
-    diamond_breve_offset = 14
+    diamond = {
+        quarter = { glyph = 79, size = 110 },
+        half  = { glyph = 79, size = 110 },
+        whole = { glyph = 79, size = 110, offset = 5 },
+        breve = { glyph = 79, size = 110, offset = 14 },
+    },
+    diamond_guitar = {
+        quarter = { glyph = 226, size = 110 },
+        half  = { glyph = 79, size = 110 },
+        whole = { glyph = 79, size = 110, offset = 5 },
+        breve = { glyph = 79, size = 110, offset = 14 },
+    },
+    x = {
+        quarter = { glyph = 192 },
+        half  = { glyph = 192 },
+        whole = { glyph = 192 },
+        breve = { glyph = 192, size = 120 },
+    },
+    triangle = {
+        -- change_shape() defaults to use "triangle_down" glyphs on "triangle" up-stems
+        -- use shape = "triangle_up" to force all up glyphs
+        -- use shape = "triangle_down" to force all down glyphs
+        quarter = { glyph = 209 },
+        half  = { glyph = 177 },
+        whole = { glyph = 177 },
+        breve = { glyph = 177 },
+    },
+    triangle_down = {
+        quarter = { glyph = 224 },
+        half  = { glyph = 198 },
+        whole = { glyph = 198 },
+        breve = { glyph = 198 },
+    },
+    triangle_up = {
+        quarter = { glyph = 209 },
+        half  = { glyph = 177 },
+        whole = { glyph = 177 },
+        breve = { glyph = 177 },
+    },
+    slash = {
+        quarter = { glyph = 243 },
+        half  = { glyph = 203 },
+        whole = { glyph = 213 },
+        breve = { glyph = 213 },
+    },
+    square = {
+        quarter = { glyph = 208 },
+        half  = { glyph = 173 },
+        whole = { glyph = 194 },
+        breve = { glyph = 221 },
+    },
+    wedge = {
+        quarter = { glyph = 108 },
+        half  = { glyph = 231 },
+        whole = { glyph = 231, offset = -14 },
+        breve = { glyph = 231, offset = -14 },
+    },
+    strikethrough = {
+        quarter = { glyph = 191 }, -- doesn't exist in Maestro
+        half  = { glyph = 191 },
+        whole = { glyph = 191 },
+        breve = { glyph = 191 },
+    },
+    circled = {
+        quarter = { glyph = 76 }, -- doesn't exist in Maestro
+        half  = { glyph = 76 },
+        whole = { glyph = 76 },
+        breve = { glyph = 76 },
+    },
+    hidden = {
+        quarter = { glyph = 202 },
+        half  = { glyph = 202 },
+        whole = { glyph = 202 },
+        breve = { glyph = 202 },
+    },
+    default = {
+        quarter = { glyph = 207 }
+    },
 }
 
--- Default to SMuFL characters for SMuFL font (without needing a config file)
+-- change to SMuFL characters for SMuFL font (without needing a config file)
 if library.is_font_smufl_font() then
```

Comment:
Let's wrap this in a function. Perhaps define both the non-SMuFL and SMuFL configs in the top-level, then have a function that returns the correct one depending on `library.is_font_smufl_font()`.

As it stands, this can introduce a subtle bug. If a script changes the font then reads noteheads, the wrong config will be used. However, if we wrap it in a function, it'll allow us to choose the correct config precisely at the time we'll need it.